### PR TITLE
Add support for the Launch Pad in VIA

### PR DESCRIPTION
**Pull request by mcmadhatter**
## Description

Added support for VIA configurator with a new via config for the launchpad.  This is as per the default keymap with as few changes as possible to add the via support. Tested on my launchpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/launchpad/keymaps/via/rules.mk [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+DEFAULT_FOLDER = launchpad/rev1
+
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/launchpad/keymaps/via/rules.mk]:
```suggestion
VIA_ENABLE = yes
```

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS  \
+)
+
+};
+
+void matrix_init_user(void) {
+
+}
```
Comment (zvecr) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
```

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _QWERTY 0
+#define _FUNC 1
+#define _LAYER2 2
+#define _LAYER3 3
```
Comment (fauxpark) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
enum layer_names {
    _QWERTY,
    _FUNC,
    _LAYER2,
    _LAYER3
};
```

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+ * |   5  |  6   |
+ * |------+------|
+ * | FUNC |  8   |
+ * `-------------'
+ */
+[_QWERTY] = LAYOUT( \
+    KC_1,      KC_2, \
+    KC_3,      KC_4, \
+    KC_5,      KC_6, \
+    MO(_FUNC), KC_8  \
```
Comment (fauxpark) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
[_QWERTY] = LAYOUT(
    KC_1,      KC_2,
    KC_3,      KC_4,
    KC_5,      KC_6,
    MO(_FUNC), KC_8
```

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ * |   Z  |  X   |
+ * |------+------|
+ * |      |  C   |
+ * `-------------'
+ */
+[_FUNC] = LAYOUT( \
+    KC_Q,    CALTDEL, \
+    KC_A,    TSKMGR, \
+    KC_Z,    KC_X, \
+    KC_TRNS, KC_C \
```
Comment (fauxpark) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
    KC_Q,    CALTDEL,
    KC_A,    TSKMGR,
    KC_Z,    KC_X,
    KC_TRNS, KC_C
```

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+ * |      |      |
+ * |------+------|
+ * |      |      |
+ * `-------------'
+ */
+[_LAYER2] = LAYOUT( \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
[_LAYER2] = LAYOUT(
    KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS
```

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+ * |      |      |
+ * |------+------|
+ * |      |      |
+ * `-------------'
+ */
+[_LAYER3] = LAYOUT( \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
[_LAYER3] = LAYOUT(
    KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS
```

Comment (mcmadhatter) [reply keyboards/launchpad/keymaps/via/keymap.c]: I was going for _minimum_ changes from what is already in the default config, but as they are non functional changes there is no harm

File: keyboards/launchpad/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+ * |   Q  |CALDEL|
+ * |------+------|
+ * |   A  |TSKMGR|
+ * |------+------|
+ * |   Z  |  X   |
+ * |------+------|
+ * |      |  C   |
+ * `-------------'
+ */
+[_FUNC] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/launchpad/keymaps/via/keymap.c]:
```suggestion
[_FUNC] = LAYOUT(
```

**Pull request merged**